packages/flutter_tools/lib/src/android/android_device.dart

```diff
@@ -648,6 +651,21 @@ class AndroidDevice extends Device {
         (int exitCode) => exitCode == 0 || allowHeapCorruptionOnWindows(exitCode));
   }
 
+  @override
+  Future<Map<String, Object>> queryRss() async {
```

Comment:
Instead of returning a `Map`, consider promoting the return type here to a class with platform agnostic fields from a base class and android-specific fields from an inherited class.

Then, instead of naming this method `queryRss` you can name it `queryBaseClass` since it's really doing a lot more than just grabbing Rss.

Reply:
I would be converting that back to JSON almost immediately though, so it would be a fairly short-lived object.

I'll wait for terry to chime in with what information we actually need from this dumpinfo first.

Reply:
When adding support for this to a new platform, it will be good to have support from the language/type-system ensuring that the expected info is filled in.

Reply:
Jonah I was hopping to get all information in the App Summary section:

Java Heap:
Native Heap:
Code::
Stack:
Graphics:
Private Other:
System:
TOTAL:

these numbers seem the most useful for Android users (or at least its what Android Studio reports in their memory profiler).

At some point, we we'll probably hookup to dumpsys battery and dumpsys network stats too.  These might be passed as a row(s) or data instead of lots of parsing (but that's in the future).  The JSON/Map seems fine for me as it will be coming via a service call.